src/js/pages/nodes/NodeDetailPage.js

```diff
@@ -0,0 +1,373 @@
+import mixin from 'reactjs-mixin';
+/* eslint-disable no-unused-vars */
+import React from 'react';
+/* eslint-enable no-unused-vars */
+import {RouteHandler} from 'react-router';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import BarChart from '../../components/charts/BarChart';
+import Chart from '../../components/charts/Chart';
+import CompositeState from '../../structs/CompositeState';
+import Config from '../../config/Config';
+import DateUtil from '../../utils/DateUtil';
+import DescriptionList from '../../components/DescriptionList';
+import HealthTab from '../../components/HealthTab';
+import InternalStorageMixin from '../../mixins/InternalStorageMixin';
+import MesosSummaryStore from '../../stores/MesosSummaryStore';
+import MesosStateStore from '../../stores/MesosStateStore';
+import NodeHealthStore from '../../stores/NodeHealthStore';
+import Page from '../../components/Page';
+import ResourceTypes from '../../constants/ResourceTypes';
+import SidePanelContents from '../../components/SidePanelContents';
+import SidePanels from '../../components/SidePanels';
+import StringUtil from '../../utils/StringUtil';
+import TabsMixin from '../../mixins/TabsMixin';
+import TaskView from '../../components/TaskView';
+import Units from '../../utils/Units';
+
+// number to fit design of width vs. height ratio
+const WIDTH_HEIGHT_RATIO = 4.5;
+
+const METHODS_TO_BIND = [
+];
+
+class NodeDetailPage extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.store_listeners = [
+      {name: 'summary', events: ['success']},
+      {name: 'state', events: ['success']},
+      {name: 'nodeHealth', events: ['nodeSuccess', 'nodeError', 'unitsSuccess', 'unitsError']}
+    ];
+
+    this.tabs_tabs = {
+      tasks: 'Tasks',
+      health: 'Health',
+      details: 'Details'
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift()
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    }, this);
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+
+    this.mountedAt = Date.now();
+
+    let node = this.getNode();
+    this.internalStorage_update({node});
+
+    if (node) {
+      NodeHealthStore.fetchNodeUnits(node.hostname);
+    }
+  }
+
+  componentWillUpdate() {
+    super.componentWillUpdate(...arguments);
+
+    if (!this.internalStorage_get().node) {
+      this.internalStorage_update({node: this.getNode()})
+    }
+  }
+
+  getBasicInfo(node) {
+    return (
+      <div className="detail-page-header">
+        <h1 className="inverse flush">
+          {node.hostname}
+        </h1>
+        {this.getSubHeader(node)}
+      </div>
+    );
+  }
+
+  getBreadcrumbs(nodeID) {
+    return (
+      <h5 className="inverse">
+        <Link className="headline emphasize" to="nodes">
+          Nodes
+        </Link>
+        <span style={{'margin': '0 15px'}}>
```

Comment:
Oh shoot that was supposed to be temporary, I'm embarrassed I left it here ahaha